jax/_src/core.py

```diff
@@ -83,6 +83,7 @@ class JaxprDebugInfo(NamedTuple):
   result_paths: tuple[str, ...]  # e.g. ('[0]', '[1]', ...)
 
 class Jaxpr:
+  _check_debug_info: bool = True
```

Comment:
I don't think this is a good idea. Please file an issue with a reproducer and we can look into fixing it.

Reply:
I will try to extract a repro from the above snippet.

Reply:
So I verified that the reproducer indeed fails with an `AssertionError`, but it's not immediately obvious what the callable being traced is.

Reply:
Can you paste the stack trace please?

Reply:
```
Traceback (most recent call last):
  File "/tmp/repro.py", line 16, in <module>
    vs = bst.transform.for_loop(run, indices)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[...]/brainstate/transform/_control.py", line 578, in for_loop
    _, ys = scan(_forloop_to_scan_fun(f),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[...]/brainstate/transform/_control.py", line 515, in scan
    stateful_fun = StatefulFunction(f).make_jaxpr(init, xs_tree.unflatten(x_avals))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[...]/brainstate/transform/_make_jaxpr.py", line 414, in make_jaxpr
    raise e
  File "[...]/brainstate/transform/_make_jaxpr.py", line 397, in make_jaxpr
    jaxpr, (out_shapes, state_shapes) = jax.make_jaxpr(
                                        ^^^^^^^^^^^^^^^
AssertionError
```